lib/parse-args.js

```diff
@@ -194,6 +194,49 @@ function buildYargs (withCommands = false) {
   return yargs
 }
 
+function loadConfigFile (path) {
+  let config = {}
+  const jsExts = ['.js', '.cjs']
+  const ymlExts = ['.yml', '.yaml']
+  const fileName = basename(path)
+
+  const ext = extname(path).toLowerCase()
+  if (jsExts.includes(ext)) {
+    config = require(path)
+  } else if (ymlExts.includes(ext)) {
+    config = require('js-yaml').load(readFileSync(path, 'utf8'))
+  } else if (ext === '.json' || fileName.slice(-2) === 'rc') {
+    config = JSON.parse(readFileSync(path, 'utf8'))
+  }
+
+  // Should the process die if none of these filename extensions are found?
```

Comment:
we should try to require() it: if there is expected situation (user passed config.ts with ts-node loaded by --require in NODE_OPTIONS, or something similar) - fine, else there is an error and process will die (which is expected in that case).

Reply:
@koshic added an error to throw if an invalid configuration file is passed with the config arguments

Reply:
So, TypeScript users can't provide their absolute valid CJS configs just because... just because we decided to don't allow that? 

Again, 'it's ok because it's better than nothing' (c) and I really appreciate your efforts, but from user perspective it looks like unreasonable restrictions. Good example is Yarn and constraints configuration file - it should be named only as 'yarn.config.js', AND must be CJS. Already migrated to ESM? Sorry. TypeScript user? Sorry.